patches/kaggle_secrets.py

```diff
@@ -57,11 +57,19 @@ def _make_post_request(self, data):
             raise BackendError('Unexpected response from the service.') from e
 
     def get_bigquery_access_token(self):
+        """
+        Calls UserSecrets service to retrieve the BigQuery access token for the current Kernel,
+        and the expiry if it is present.
+        Example usage:
+            client = UserSecretsClient()
+            self.token, self.expiry = client.get_bigquery_access_token()
+        """
         request_body = {
             'Target': self.BIGQUERY_TARGET_VALUE
         }
         response_json = self._make_post_request(request_body)
         if 'secret' not in response_json:
             raise BackendError(
                 'Unexpected response from the service.')
-        return response_json['secret']
+        # Optionally return expiry if it is set.
+        return response_json['secret'], response_json.get('expiresInSeconds')
```

Comment:
Sorry, if this was confusing over Slack: before assigning the result to `self.expiry`, we should add the value of `expiresInSeconds` to `now`, similarly to how they do it here:
https://github.com/googleapis/google-auth-library-python/blob/master/google/oauth2/_client.py#L64.

The result should be of type `datetime`, e.g. `nowUtc + datetime.timedelta(seconds=expires_in)` as in this doc:
https://github.com/googleapis/google-auth-library-python/blob/master/google/auth/credentials.py#L49

Reply:
Basically, "when the token expires and is no longer valid" is now + expiration in seconds (which is usually 1 hour).

Reply:
Is that what the auth service returns to the Web Tier? Some time may have already passed since the token was issued to the time the client nowUtc + datetime.timedelta(seconds=expires_in), so it won't really be accurate. Is it not possible for the Web Tier to return an exiration time?

Reply:
This is what is passed by Google Auth service back when we refresh access token.
Yes, there could be a delay of a couple of seconds, or something like that, but that is OK. 

This value gets adjusted later by the client library by 5 minutes, to make sure the refresh request happens more or less close to the expiration. See this line in `expired` method:
https://github.com/googleapis/google-auth-library-python/blob/master/google/auth/credentials.py#L63

Reply:
Done.